DirectProgramming/C++SYCL/DenseLinearAlgebra/guided_jacobiCudaGraphs_SYCLMigration/03_sycl_migrated_optimized/include/dpct/kernel.hpp

```diff
@@ -0,0 +1,459 @@
+//==---- kernel.hpp -------------------------------*- C++ -*----------------==//
+//
+// Copyright (C) Intel Corporation
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __DPCT_KERNEL_HPP__
+#define __DPCT_KERNEL_HPP__
+
+#include <sycl/sycl.hpp>
+#ifdef _WIN32
+#include <unordered_set>
+#include <windows.h>
+#else
+#include <dlfcn.h>
+#endif
+
+#if defined(__has_include) && __has_include(<filesystem>)
+#include <filesystem>
+#elif defined(__has_include) && __has_include(<experimental/filesystem>)
+#include <experimental/filesystem>
+#else
+#error "SYCLomatic runtime requires C++ filesystem support"
+#endif
+
+#include <random>
+
```

Comment:
@Shwetha-Selma Can you add ``#include <fstream>``? The 2024.0 compiler does not include that header file implicitly, so we will need to add it explicitly. This will resolve the compile error below.

Error I am getting in 2024.0:
``/oneAPI-samples/DirectProgramming/C++SYCL/DenseLinearAlgebra/guided_simpleCudaGraphs_SYCLMigration/02_sycl_migrated/include/dpct/kernel.hpp:119:18: error: implicit instantiation of undefined template 'std::basic_ofstream<char>'``

Reply:
Sure @jimmytwei, I have updated sample with #include <fstream>.